### DISCUSSION TITLE
Use `nvim-treesitter` for normal tree-sitter CLI

### DISCUSSION THREAD

**Discussion by crides** [nvim-treesitter/nvim-treesitter#2904] [Q&A]
This is not really related to `nvim-treesitter`, but it seems like the normal tree sitter CLI uses a different set/layout of parsers. Since obviously `nvim-treesitter` can use the format in this repo, is there a way to let the CLI also use the parsers downloaded by this repo directly?

**Comment by kyazdani42** [comment]
@theHamsta any idea ? I have no clue about this

**Comment by theHamsta** [comment]
The CLI can is meant to run directly in the parser repo. So you could just check out the repo like we do and then let the CLI run in the downloaded folder. This is also what we do in `:TSInstallFromGrammar`. For `tree-sitter highlight`, I think you can also have the language in a special subfolder in you home directory but I'm not an expert in that. Reading the source code of `tree-sitter` the parsers should go to `~/.cache/tree-sitter/lib/<lang>[.debug].so` (but I never found the shared objects there or in this folder (with additional node bindings) when doing a `npm build`
```
➜  tree-sitter-glsl git:(master) ✗ ldd build/Release/tree_sitter_glsl_binding.node 
        linux-vdso.so.1 (0x00007ffc780d5000)
        libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f67df473000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f67df38f000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f67df375000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f67df14d000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f67df739000)
➜  tree-sitter-glsl git:(master) ✗ nm build/Release/tree_sitter_glsl_binding.node
```
The tree-sitter CLI usually directly loads the language from the source files and uses it's internal cache to decide whether a recompilation is needed.

**Reply by NonlinearFruit** [reply]
Related discussion on nvim-treesitter in tree-sitter ([discussion](https://github.com/tree-sitter/tree-sitter/discussions/2717))
